pkgs/development/rocm-modules/6/default.nix

```diff
@@ -24,69 +21,88 @@ let
       inherit (self) llvm;
       pyPackages = python3Packages;
       openmpi-orig = openmpi;
+      rocmClangStdenv = llvm.rocmClangStdenv;
     in
     {
-      inherit rocmGpuArches;
+      inherit rocmClangStdenv;
+      stdenv = rocmClangStdenv;
       buildTests = false;
       buildBenchmarks = false;
-      stdenv = llvm.rocmClangStdenv;
 
       rocmPath = self.callPackage ./rocm-path { };
       rocmUpdateScript = self.callPackage ./update.nix { };
 
       ## ROCm ##
       llvm = recurseIntoAttrs (
         callPackage ./llvm/default.nix {
-          inherit (self) rocm-device-libs rocm-runtime;
+          # rocm-device-libs is used for .src only
+          # otherwise would cause infinite recursion
+          inherit (self) rocm-device-libs;
         }
       );
       inherit (self.llvm) rocm-merged-llvm clang openmp;
 
-      rocm-core = self.callPackage ./rocm-core { };
-      amdsmi = pyPackages.callPackage ./amdsmi {
-        inherit (self) rocmUpdateScript;
+      rocm-core = self.callPackage ./rocm-core {
+        inherit stdenv;
       };
```

Comment:
Shouldn’t need to `inherit` it if it’s in the scope. (Applies to a bunch of stuff in this file.)

Reply:
renamed but it's still not very clear.